### ISSUE TITLE
Corresponding author email verification

### ISSUE THREAD

**Issue by hdrury1** [elifesciences/elife-xpub#272]
### User Story

_As a submitter I want a notification ribbon in order to know whether or not the corresponding author has accepted their responsibility._

### Product requirements/Acceptance criteria

Functional:

- When the submitter edits the email in the field on page 1 of the submission wizard and it no longer matches the email associated with the submitter's ORCiD account, a ribbon is displayed at the top of the page after the submitter hits 'Next'.
- This ribbon remains at the top of every page of the wizard until the corresponding author takes an action to validate themselves as such.
- The submitter can use the 'resend' link within the ribbon to send another validation email to the nominated corresponding author.
- If the submitter resends the validation email, the ribbon updates to indicate this.
- The presence of the ribbon should not stop the aubmitter from completing the wizard.

### Relevant links to design(s)
[Notification ribbon](https://projects.invisionapp.com/share/CDLI0R5KBF6#/screens) -  Out of date: notification ribbon to be removed from submission wizard & (possibly) displayed on dashboard instead

[Email to be received by corresponding author](https://github.com/elifesciences/elife-xpub/files/2105192/corresponding-author-email.pdf)

[Confirmation screen (with log in button)](https://user-images.githubusercontent.com/17023702/41456174-8b2d6a60-7077-11e8-86bd-dc20bd7e6533.png)

[Confirmation screen (without log in button)](https://user-images.githubusercontent.com/17023702/42327368-733f570a-8063-11e8-92bd-34fe7858a3d0.png)

[Decline screen](https://user-images.githubusercontent.com/17023702/41456190-958a838a-7077-11e8-8617-1540504485d5.png)

### Tasks
When we thought the email would be sent on clicking 'Next' from step 1 of the wizard:
- [x] Send verification email to corresponding author if they are not the submitter #154 
    - change backend code to differentiate between autosaving & clicking Next
    - alter schema to include `ManuscriptPerson` & `UserAlias`
    - check if there is an existing user in the system matching the author's email address
    - if not, send email using `@pubsweet/component-send-email` component (for now?)
- [x] Create notification ribbon component (functionality to be added) #228 
- [ ] Constantly display notification ribbon on all wizard pages by default #285 

Change in requirements:
- [x] Email correpsonding author on 'Submit' instead of 'Next' #310 

**Note: more product, design & technical decisions to be made before the following work can be completed**

Build Corresponding Author's "user journey":
- [ ] Data model support for changing Corresponding Author from `User Alias` to `User` #311 
- [ ] Fetch Orcid user details on login #309 
- [ ] Create corresponding author email & associated "thank you" pages (Confirm & Deny) #204 
- [ ] Token verification
- [ ] Mutate the manuscript's corresponding author object
- [ ] Connect the corresponding author's confirm/deny state, to the show/hide of the notification ribbon
- [ ] Handle unhappy paths i.e. show error pages if/when token is invalid
- [ ] Allow submitter to resend verification email to author #286
- [ ] Allow corresponding author to log in with orcid #298

**Comment by jsms90** [comment]
@hdrury1 We seem to be using this issue as a kind of meta-issue, with links to all the sub-issues which relate to the notification ribbon & its functionality. In fact, it's more than the notification ribbon, which is only the visual representation of this feature, from the perspective of a submitter. This feature also encompasses the journey that a corresponding author goes through (the email itself, the Thank you pages, etc).

Since we seemed to be linking from this issue to all the others anyway, I've carried on that pattern & changed the title of this issue to reflect that this is where we are keeping track of the whole email verification feature.

Hopefully we can use this issue to give you greater visibility over all the steps involved in completing this feature & how far through each one we have gotten?
